xdem/coreg/base.py

```diff
@@ -313,6 +314,33 @@ def _preprocess_coreg_fit(
     ):
         raise ValueError("Input elevation data should be a raster, an array or a geodataframe.")
 
+    # If inlier_mask has not the same shape of input dems, reproject it into an into dem # todo
```

Comment:
This code only applies when the inputs are rasters, so should be moved to `_preprocess_coreg_fit_raster_raster` called directly below.